src/Sulu/Component/Content/Types/Rlp/Mapper/PhpcrMapper.php

```diff
@@ -31,62 +33,71 @@ class PhpcrMapper extends RlpMapper
     private $sessionManager;
 
     /**
-     * @param SessionManagerInterface $sessionManager
+     * @var DocumentManagerInterface
      */
-    public function __construct(SessionManagerInterface $sessionManager)
-    {
+    private $documentManager;
+
+    /**
+     * @var DocumentInspector
+     */
+    private $documentInspector;
+
+    public function __construct(
+        SessionManagerInterface $sessionManager,
+        DocumentManagerInterface $documentManager,
+        DocumentInspector $documentInspector
+    ) {
         $this->sessionManager = $sessionManager;
+        $this->documentManager = $documentManager;
+        $this->documentInspector = $documentInspector;
     }
 
     /**
-     * creates a new route for given path.
-     *
-     * @param NodeInterface $contentNode  reference node
-     * @param string        $path         path to generate
-     * @param string        $webspaceKey  key of webspace
-     * @param string        $languageCode
-     * @param string        $segmentKey
+     * {@inheritdoc}
      */
-    public function save(NodeInterface $contentNode, $path, $webspaceKey, $languageCode, $segmentKey = null)
+    public function save(ResourceSegmentBehavior $document)
     {
-        $routes = $this->getWebspaceRouteNode($webspaceKey, $languageCode, $segmentKey);
+        $webspaceKey = $this->documentInspector->getWebspace($document);
+        $locale = $this->documentInspector->getLocale($document);
+        $segmentKey = null;
+        $webspaceRouteRootPath = $this->getWebspaceRouteNodeBasePath($webspaceKey, $locale, $segmentKey);
 
-        // check if route already exists
-        if ($this->checkResourceLocator($routes, $path, $contentNode)) {
-            return;
-        }
+        try {
+            $routeNodePath = $this->loadByContent(
+                $this->documentInspector->getNode($document),
+                $webspaceKey,
+                $locale,
+                $segmentKey
+            );
 
-        // create root recursive
-        $routePath = explode('/', ltrim($path, '/'));
-        $node = $routes;
-        foreach ($routePath as $path) {
-            if ($path != '') {
-                if ($node->hasNode($path)) {
-                    $node = $node->getNode($path);
-                } else {
-                    $node = $node->addNode($path, 'nt:unstructured');
-                }
+            if ($routeNodePath === '/') {
+                return;
             }
+
+            $routeDocument = $this->documentManager->find($webspaceRouteRootPath . $routeNodePath);
+            $routeDocumentPath = $webspaceRouteRootPath . $routeNodePath;
```

Comment:
But that's not even the path of the document I am using here. It is the relative path of the current route node (returned by `loadByContent`).